bubble_sort.rb

```diff
@@ -0,0 +1,57 @@
+def bubble_sort(array)
+  # prints the initial unsorted array
+  puts "Un-sorted array: #{array}"
+  arr = array.length - 1
+  loop do
+    # created a boolean variable, default is false.
+    swap = false
+    # loop through the array
+    (0...arr).each do |i|
+      # created two variables for the first number and the one after as 'i'
+      # so it keeps checking every number next to eachother.
+      n_one = array[i]
+      n_two = array[i + 1]
+
+      if n_one > n_two
+        # if the first number is greater than the one after, swap.
+        array[i], array[i + 1] = array[i + 1], array[i]
+        # swap becomes true when a swap has happened
+        swap = true
+      end
+      next unless
+        n_two > n_one
+    end
+    # the loop breaks when no swap has happened
+    break if swap == false
+  end
+  "Sorted array: #{array}"
+end
```

Comment:
- [ ] You should just return the sorted array here. Using string interpolation is causing the tests to fail although the array is sorted correctly. I have attached below the failing test: 
![Screen Shot 2020-08-25 at 7 07 45 PM](https://user-images.githubusercontent.com/48804258/91205694-431cc380-e706-11ea-96bf-06d1862346e2.png)